modules/volunteers/sitesupport/hourTable.js

```diff
@@ -110,6 +36,90 @@ export default {
     }
   },
   methods: {
+    load() {
+      this.records = null;
+      this.totalHours = 0;
+      this.totalSpentHours = 0;
+      this.hours = [];
+      this.totalVolunteers = null;
+
+      showSpinner();
+      this.summeryColumns = [
+        {value:'name', title:'Department Worked', source:null},
+
+        {value:'entry_count', title:'Total Entries', source:this.hours},
+        {value:'volunteer_count', title:'Total Volunteers', source:this.hours},
+        {value:'total_hours', title:'Total Hours', source: this.hours}
+      ];
+      this.detailColumns = [
+        {value:'name', title:'Department Worked', source:null},
+        {value:'end', title:'Ended', source: this.hours},
+        {value:'hours', title:'Hours', source: this.hours}
+      ];
+
+      apiRequest('GET', '/department','max_results=all')
+        .then((response) => {
+          const result = JSON.parse(response.responseText);
+          this.departments = [];
+          result.data.forEach((entry) => {
+            this.departments.push(entry);
+          });
+
+          this.departments.sort((a,b) => (a.name > b.name) ? 1 : -1);
+
+          if (this.$parent.userId != null) {
+            this.columns = this.detailColumns;
+            this.records = this.hours;
+            apiRequest('GET', '/member/' + this.$parent.userId + '/volunteer/hours','max_results=all')
+              .then((response) => {
+                const result = JSON.parse(response.responseText);
+                result.data.forEach((entry) => {
+                  this.hours.push(entry);
+                })
```

Comment:
Would this be cleaner?

`this.hours = result.data;`

Does `result.data` come back with more than what we want to actually track here?

Reply:
Something funky happens here...
So above you see `this.records = this.hours`  which should be assignment by reference.
if we do the whole `this.hours.push` thing, then `this.records` is correct. However if we just assign `this.hours` to `result.data` then `this.records` is not updated and remains empty.

Reply:
That seems weird. I can explore more into that as I dive into Vue, it could just be a quirk of the framework or something.

Another clean option might be to do something like

`this.hours.push(...result.data);` or `this.hours = [...result.data];`